apps/verticalhome/js/more_apps_navigation.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+(function (exports) {
+
+  var MoreAppsNavigation = {
+    controls: null,
+    selectedElemIndex: 0,
+    selectElement: null,
+
+    init: function () {
+      this.controls = app.grid.getItems();
+      this.reset();
+    },
+
+    reset: function () {
+      if (this.selectElement) {
+        this.selectElement.classList.remove('selected');
+      }
+      this.selectedElemIndex = 0;
+      this.selectElement = this.controls[this.selectedElemIndex].element;
+      this.selectElement.classList.add('selected');
+      this.clearNavigationEvents();
+      window.addEventListener('keydown', this);
+    },
+
+    clearNavigationEvents: function () {
+      window.removeEventListener('keydown', this);
+    },
+
+    handleEvent: function (e) {
+      var deltaIndex = 0;
+      e.preventDefault();
+      switch (e.key) {
+        case 'ArrowUp':
+          deltaIndex = -2;
+          this.changeSelectElem(deltaIndex);
+          if (!this.isVisible(this.selectElement)) {
+            this.selectElement.scrollIntoView({behavior:"smooth", block: "start"});
+          }
+          break;
+        case 'ArrowDown':
+          deltaIndex = 2;
+          this.changeSelectElem(deltaIndex);
+          if (!this.isVisible(this.selectElement)) {
+            this.selectElement.childNodes[0].scrollIntoView({behavior:"smooth", block: "end"});
+          }
+          break;
+        case 'ArrowLeft':
+          deltaIndex = -1;
+          this.changeSelectElem(deltaIndex);
+          if (!this.isVisible(this.selectElement)) {
+            this.selectElement.scrollIntoView({behavior:"smooth", block: "start"});
+          }
+          break;
+        case 'ArrowRight':
+          deltaIndex = 1;
+          this.changeSelectElem(deltaIndex);
+          if (!this.isVisible(this.selectElement)) {
+            this.selectElement.childNodes[0].scrollIntoView({behavior:"smooth", block: "end"});
+          }
+          break;
+        case 'Accept':
+          window.removeEventListener('keydown', this);
+          this.selectElement.click();
+          break;
+        case 'BrowserBack':
+        case 'Backspace':
+          app.backToMainScreen();
+          break;
+      }
+    },
+
+    changeSelectElem: function (deltaIndex) {
+      this.selectElement.classList.remove('selected');
+      var temp = this.selectedElemIndex + deltaIndex;
+      if (temp > -1 && temp < this.controls.length) {
+        this.selectedElemIndex = temp;
+      }
+      this.selectElement = this.controls[this.selectedElemIndex].element;
+      this.selectElement.classList.add('selected');
+    },
+
+    isVisible: function (element) {
```

Comment:
Do you really need this?  If so, dosn't the grid component already do it for you somehow?  This doesn't seem like the best way to do it.  In particular, I suspect you want getClientBoundingRect rather than querying the individual rects.